enhancements/authentication/invalid-certs.md

```diff
@@ -0,0 +1,200 @@
+---
+title: invalid-certs
+authors:
+  - "@s-urbaniak"
+reviewers:
+  - "@sttts"
+  - "@stlaz"
+  - "@deads2k"
+approvers:
+  - "@mfojtik"
+creation-date: 2022-12-08
+last-updated: 2022-12-08
+---
+
+# Detecting invalid CA certificates
+
+## Summary
+
+OpenShift 4.10 is going to be rebased against Kubernetes 1.23. This requires using Go 1.17.
+However, starting with Go 1.17 support for invalid certificates is going to be removed, see https://go.dev/doc/go1.17.
+Formally, the temporary `GODEBUG=x509ignoreCN=0` flag has been removed.
+This implies that starting from OpenShift 4.10 invalid certificates will not be trusted any more as they will fail verification.
+
+Example:
+
+Given the following certificate:
+```
+Certificate:
+    Data:
+        ...
+        Subject: CN=foo-domain.com
+        X509v3 extensions:
+            X509v3 Key Usage: critical
+                Digital Signature, Key Encipherment
+            X509v3 Extended Key Usage: 
+                TLS Web Server Authentication
+            X509v3 Basic Constraints: critical
+                CA:FALSE
+```
+
+Verification against the `foo-domain.com` hostname of such certificate will fail with the following error in Go 1.17:
+```
+x509: certificate relies on legacy Common Name field, use SANs instead
+```
+
+Verification of server certificates is executed during TLS client handshakes,
```

Comment:
You probably mean during "TLS Handshake procedure" or when a client verified "Server Hello".